File: src/ikj/main/ioke/lang/Ground.java
/*
 * See LICENSE file in distribution for copyright and licensing information.
 */
package ioke.lang;

import java.util.ArrayList;
import java.util.HashMap;

import ioke.lang.exceptions.ControlFlow;

/**
 * The Ground serves the same purpose as the Lobby in Self and Io.
 * This is the place where everything is evaluated.
 *
 * @author <a href="mailto:ola.bini@gmail.com">Ola Bini</a>
 */
public class Ground {
    public static void init(IokeObject iokeGround, IokeObject ground) throws ControlFlow {
        Runtime runtime = ground.runtime;
        iokeGround.setKind("IokeGround");
        ground.setKind("Ground");
        iokeGround.registerCell("Base", runtime.base);
        iokeGround.registerCell("DefaultBehavior", runtime.defaultBehavior);
        iokeGround.registerCell("IokeGround", runtime.iokeGround);
        iokeGround.registerCell("Ground", runtime.ground);
        iokeGround.registerCell("Origin", runtime.origin);
        iokeGround.registerCell("System", runtime.system);
        iokeGround.registerCell("Runtime", runtime.runtime);
        iokeGround.registerCell("Text", runtime.text);
        iokeGround.registerCell("Symbol", runtime.symbol);
        iokeGround.registerCell("Number", runtime.number);
        iokeGround.registerCell("nil", runtime.nil);
        iokeGround.registerCell("true", runtime._true);
        iokeGround.registerCell("false", runtime._false);
        iokeGround.registerCell("Arity", runtime.arity);
        iokeGround.registerCell("Method", runtime.method);
        iokeGround.registerCell("DefaultMethod", runtime.defaultMethod);
        iokeGround.registerCell("JavaMethod", runtime.javaMethod);
        iokeGround.registerCell("LexicalBlock", runtime.lexicalBlock);
        iokeGround.registerCell("DefaultMacro", runtime.defaultMacro);
        iokeGround.registerCell("LexicalMacro", runtime.lexicalMacro);
        iokeGround.registerCell("DefaultSyntax", runtime.defaultSyntax);
        iokeGround.registerCell("Mixins", runtime.mixins);
        iokeGround.registerCell("Restart", runtime.restart);
        iokeGround.registerCell("List", runtime.list);
        iokeGround.registerCell("Dict", runtime.dict);
        iokeGround.registerCell("Set", runtime.set);
        iokeGround.registerCell("Range", runtime.range);
        iokeGround.registerCell("Pair", runtime.pair);
        iokeGround.registerCell("DateTime", runtime.dateTime);
        iokeGround.registerCell("Message", runtime.message);
        iokeGround.registerCell("Call", runtime.call);
        iokeGround.registerCell("Condition", runtime.condition);
        iokeGround.registerCell("Rescue", runtime.rescue);
        iokeGround.registerCell("Handler", runtime.handler);
        iokeGround.registerCell("IO", runtime.io);
        iokeGround.registerCell("FileSystem", runtime.fileSystem);
        iokeGround.registerCell("Regexp", runtime.regexp);
        iokeGround.registerCell("JavaGround", runtime.javaGround);

        iokeGround.registerMethod(runtime.newJavaMethod("will return a text representation of the current stack trace", 
                                                    new JavaMethod.WithNoArguments("stackTraceAsText") {
                                                        @Override
                                                        public Object activate(IokeObject method, IokeObject context, IokeObject message, Object on) throws ControlFlow {
                                                            getArguments().getEvaluatedArguments(context, message, on, new ArrayList<Object>(), new HashMap<String, Object>());

                                                            return context.runtime.newText("");
                                                        }}));
    }
}// Ground

File: src/ikc/main/Ioke.Lang/Ground.cs
namespace Ioke.Lang {
    using System.Collections;
    using System.Collections.Generic;

    using Ioke.Lang.Util;

    public class Ground {
        public static void Init(IokeObject iokeGround, IokeObject ground) {
            Runtime runtime = ground.runtime;
            iokeGround.Kind = "IokeGround";
            ground.Kind = "Ground";
            iokeGround.RegisterCell("Base", runtime.Base);
            iokeGround.RegisterCell("DefaultBehavior", runtime.DefaultBehavior);
            iokeGround.RegisterCell("IokeGround", runtime.IokeGround);
            iokeGround.RegisterCell("Ground", runtime.Ground);
            iokeGround.RegisterCell("Origin", runtime.Origin);
            iokeGround.RegisterCell("System", runtime.System);
            iokeGround.RegisterCell("Runtime", runtime._Runtime);
            iokeGround.RegisterCell("Text", runtime.Text);
            iokeGround.RegisterCell("Symbol", runtime.Symbol);
            iokeGround.RegisterCell("Number", runtime.Number);
            iokeGround.RegisterCell("nil", runtime.nil);
            iokeGround.RegisterCell("true", runtime.True);
            iokeGround.RegisterCell("false", runtime.False);
            iokeGround.RegisterCell("Arity", runtime.Arity);
            iokeGround.RegisterCell("Method", runtime.Method);
            iokeGround.RegisterCell("DefaultMethod", runtime.DefaultMethod);
            iokeGround.RegisterCell("NativeMethod", runtime.NativeMethod);
            iokeGround.RegisterCell("LexicalBlock", runtime.LexicalBlock);
            iokeGround.RegisterCell("DefaultMacro", runtime.DefaultMacro);
            iokeGround.RegisterCell("LexicalMacro", runtime.LexicalMacro);
            iokeGround.RegisterCell("DefaultSyntax", runtime.DefaultSyntax);
            iokeGround.RegisterCell("Mixins", runtime.Mixins);
            iokeGround.RegisterCell("Restart", runtime.Restart);
            iokeGround.RegisterCell("List", runtime.List);
            iokeGround.RegisterCell("Dict", runtime.Dict);
            iokeGround.RegisterCell("Set", runtime.Set);
            iokeGround.RegisterCell("Range", runtime.Range);
            iokeGround.RegisterCell("Pair", runtime.Pair);
            iokeGround.RegisterCell("DateTime", runtime.DateTime);
            iokeGround.RegisterCell("Message", runtime.Message);
            iokeGround.RegisterCell("Call", runtime.Call);
            iokeGround.RegisterCell("Condition", runtime.Condition);
            iokeGround.RegisterCell("Rescue", runtime.Rescue);
            iokeGround.RegisterCell("Handler", runtime.Handler);
            iokeGround.RegisterCell("IO", runtime.Io);
            iokeGround.RegisterCell("FileSystem", runtime.FileSystem);
            iokeGround.RegisterCell("Regexp", runtime.Regexp);

            iokeGround.RegisterMethod(runtime.NewNativeMethod("will return a text representation of the current stack trace", 
                                                              new NativeMethod.WithNoArguments("stackTraceAsText",
                                                                                               (method, context, message, on, outer) => {
                                                                                                   outer.ArgumentsDefinition.GetEvaluatedArguments(context, message, on, new SaneArrayList(), new SaneDictionary<string, object>());
                                                                                                   return context.runtime.NewText("");
                                                                                               })));
        }
    }
}
